packages/app/src/components/Synchronization/SynchronizationAPIProvider.tsx

```diff
@@ -15,34 +20,81 @@ type SynchronizationAPIProviderProps = PropsWithChildren<{
   accessToken: string;
 }>;
 
+export interface SynchronizationConfig {
+  isAuthorized: boolean;
+  login: () => void;
+}
+
+export const SynchronizationContext = createContext<SynchronizationConfig>({
+  isAuthorized: false,
+  login: () => {
+    /** nop */
+  },
+});
+
 export const SynchronizationAPIProvider = ({
   accessToken,
   children,
 }: RouteComponentProps & SynchronizationAPIProviderProps) => {
   const urlPrefix = useApiPrefix();
-  const [isAuthorized, setIsAuthorized] = React.useState(false);
+  const [isAuthorized, setIsAuthorized] = useState(false);
+  const [authUrl, setAuthUrl] = useState<string>();
+  const [client, setClient] = useState<SynchronizationClient>();
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (!accessToken) {
       return;
     }
+    setClient(new SynchronizationClient(urlPrefix, accessToken));
+  }, [accessToken, urlPrefix]);
 
-    const client = new SynchronizationClient(urlPrefix, accessToken);
-    client
-      .getAuthorization(window.location.href)
-      .then(({ authorizationInformation }) => {
-        if (
-          !authorizationInformation?.isUserAuthorized &&
-          authorizationInformation?._links?.authorizationUrl?.href
-        ) {
-          return navigate(
-            authorizationInformation._links.authorizationUrl.href
-          );
+  useEffect(() => {
+    if (client) {
+      client
+        .getAuthorization(window.location.href)
+        .then(({ authorizationInformation }) => {
+          if (
+            !authorizationInformation?.isUserAuthorized &&
+            authorizationInformation?._links?.authorizationUrl?.href
+          ) {
+            setAuthUrl(authorizationInformation._links.authorizationUrl.href);
+          } else {
+            setIsAuthorized(true);
+          }
+        })
+        .catch(console.error);
```

Comment:
This will rerun each time the token is refreshed, should we `setIsAuthorized(false)` when an error occurs ? (If we don't, should we stop tyring to update this value once it was set to true at least once ? (so on line 52: `if(client && !isAuthorized) {` ?

Reply:
made the login synchronous and added a negative toast should the user close the login window without being authorized